internal/version/version.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright The Helm Authors, SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"flag"
+	"runtime"
+)
+
+var (
+	// version is the current version of Hypper.
+	// Update this whenever making a new release.
+	// The version is of the format Major.Minor.Patch[-Prerelease][+BuildMetadata]
+	//
+	// Increment major number for new feature additions and behavioral changes.
+	// Increment minor number for bug fixes and performance enhancements.
+	version = "v0.0.1"
```

Comment:
silly question, does goreleaser use this?

Reply:
What goreleaser does is sets the override for this using the ldflags like we do here on the Makefile[0]

Of course, I cant add that to the goreleaser until the flags are set here...I think?? So this is actually blocking the goreleaser PR and I just noticed it!


[0] https://github.com/rancher-sandbox/hypper/pull/98/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R22

Reply:
its easier to see it run:

```

# itxaka @ zeus in ~/projects/hypper on git:add_version o [15:39:35] C:128
$ git tag v0.0.12

# itxaka @ zeus in ~/projects/hypper on git:add_version o [15:39:39] 
$ make build
go fmt ./...
internal/version/version.go
go vet ./...
go build  -tags '' -ldflags '-w -s -X github.com/rancher-sandbox/hypper/internal/version.version=v0.0.12 -X github.com/rancher-sandbox/hypper/internal/version.gitCommit=0f51f331425a55878bd69872d8a1ee42bef5441e' -o '/home/itxaka/projects/hypper/bin'/"hypper" ./cmd/hypper

# itxaka @ zeus in ~/projects/hypper on git:add_version x [15:39:50] 
$ bin/hypper version                                  
version.BuildInfo{Version:"v0.0.12", GitCommit:"0f51f331425a55878bd69872d8a1ee42bef5441e", GoVersion:"go1.16"}

```

In this case the makefile is setting `X github.com/rancher-sandbox/hypper/internal/version.version=v0.0.12` so the end binary has that version. Goreleaser will do the same by setting the same flag on build

Reply:
many thanks for the extended explanation and example!